File: Python/exercise3.py
def exercise3(l):
    """
    Exercise 3 - All Sublists:
    Returns all the possible sublists within an input Array list.

    Parameters:
    - l (Array): The input list which sublists are extracted from.

    Returns:
    - Array: An array list containing all possible sublists.
    """
    sublists = [[]]
    for i in range(len(l)):  # Outer for loop iterates through the Array
        for j in range(i, len(l)):  # Inner for loop iterates from the current index to the end
            sublists.append(l[i: j + 1])  # Slices the list and adds them to the sublists array
    return sublists

File: JS/exercise3.js
/**
 * Exercise 3 - All Sublists: Returns all the possible sublists within a input Array list.
 * @param {Array} l - The input list which sublists are extracted from.
 * @returns {Array} - An array list containing all possible sublists.
 */

function exercise3(l) {
    let sublists = [[]];
    // Outer for loop iterates through the Array
    for (let i = 0; i < l.length; i++) {
        for (let j = i; j < l.length; j++) {
            // Inner for loop iterates the end position index from current position to the end
            sublists.push(l.slice(i, j + 1)); // Slices the list and adds them to the sublists array
        }
    }
    return sublists;
}
